source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTableView.mm

```diff
@@ -0,0 +1,114 @@
+//
+//  ACRTableView.m
+//  ACRTableView
+//
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+
+#import "ACRTableView.h"
+#import "ACOBaseCardElementPrivate.h"
+#import "ACOHostConfigPrivate.h"
+#import "ACRTableRow.h"
+#import "Table.h"
+#import "TableCell.h"
+#import "TableColumnDefinition.h"
+#import "TableRow.h"
+
+@implementation ACRTableView {
+    NSMutableArray<ACRColumnDefinition *> *_columnDefinitions;
+}
+
+- (instancetype)init:(ACOBaseCardElement *)acoElement
+           viewGroup:(UIView<ACRIContentHoldingView> *)viewGroup
+            rootView:(ACRView *)rootView
+              inputs:(NSMutableArray *)inputs
+          hostConfig:(ACOHostConfig *)acoConfig;
+{
+    self = [super init];
+    if (self) {
+        std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
+        std::shared_ptr<Table> table = std::dynamic_pointer_cast<Table>([acoElement element]);
+        [viewGroup addArrangedSubview:self];
+        _columnDefinitions = [[NSMutableArray alloc] init];
+        [self defineColumnDefinitions:table];
+        [self buildRowView:table rootView:rootView inputs:inputs hostConfig:acoConfig];
+    }
+    return self;
+}
+
+- (void)defineColumnDefinitions:(const std::shared_ptr<Table> &)table
+{
+    CGFloat totalRelativeWidth = 0, totalPixelWidth = 0;
+    NSInteger i = 0;
+    for (const auto &columnDefinition : table->GetColumns()) {
+        auto optionalNumericValue = columnDefinition->GetWidth();
+        if (optionalNumericValue.has_value()) {
+            totalRelativeWidth += *optionalNumericValue;
+        } else if (auto optionalPixelValue = columnDefinition->GetWidth(); optionalPixelValue.has_value()) {
+            totalPixelWidth += *optionalPixelValue;
+        }
+        i++;
+    }
+
+    // if total relative width is zero, dividend will be zero, and setting the value to 1
+    // ensures the relative width come out as zero
+    totalRelativeWidth = (totalRelativeWidth == 0) ? 1 : totalRelativeWidth;
```

Comment:
Thanks for the suggestion, but I will do so when there is more than one use case. This is a particular use case. In this case, it makes sense to take the limit as 0 instead of the infinity.